libraries/utils/src/render-lines-on-sp.ts

```diff
@@ -63,58 +70,66 @@ export class ScatterPlotLinesRenderer {
 
 
     renderLines(): void {
-        const filterBitset = this.sp.dataFrame.filter;
-        const linesPerPairCounter: { [key: string]: number } = {};
         for (let i = 0; i < this.lines.from.length; i++) {
-            if (filterBitset.get(this.lines.from[i]) && filterBitset.get(this.lines.to[i])) {
-                const pointFrom = this.sp.worldToScreen(this.xAxisValues[this.lines.from[i]], this.yAxisValues[this.lines.from[i]]);
-                const aX = pointFrom?.x;
-                const aY = pointFrom?.y;
-                const pointTo = this.sp.worldToScreen(this.xAxisValues[this.lines.to[i]], this.yAxisValues[this.lines.to[i]]);
-                const bX = pointTo?.x;
-                const bY = pointTo?.y;
-                if (aX && aY && bX && bY) {
-                    const pairStr = this.getStringPair(this.lines.from[i], this.lines.to[i]);
-                    !linesPerPairCounter[pairStr] ? linesPerPairCounter[pairStr] = 1 : linesPerPairCounter[pairStr]++;
-                    const line = new Path2D();
-                    this.paths[i] = line;
-                    line.moveTo(aX!, aY!);
-                    const color = this.lines.colors?.[i] ? this.lines.colors?.[i] : '0,128,0';
-                    const opacity = this.lines.opacities?.[i] ? this.lines.opacities?.[i] : 1;
-                    this.ctx.strokeStyle = `rgba(${color},${opacity})`;
-                    this.ctx.lineWidth = this.lines.widths?.[i] ? i === this.currentLineIdx ? this.lines.widths?.[i] + 2 :
+            let pointFromIdx = this.lines.from[i];
+            let pointToIdx = this.lines.to[i];
+            let changedPoints = false;
+            if (pointFromIdx > pointToIdx) {
+                pointFromIdx = this.lines.to[i];
+                pointToIdx = this.lines.from[i];
+                changedPoints = true;
+            }
+            const pointFrom = this.sp.worldToScreen(this.xAxisCol.get(pointFromIdx), this.yAxisCol.get(pointFromIdx));
+            const aX = pointFrom?.x;
+            const aY = pointFrom?.y;
+            const pointTo = this.sp.worldToScreen(this.xAxisCol.get(pointToIdx), this.yAxisCol.get(pointToIdx));
+            const bX = pointTo?.x;
+            const bY = pointTo?.y;
+            if (aX && aY && bX && bY) {
+                const line = new Path2D();
+                this.paths[i] = line;
+                line.moveTo(aX!, aY!);
+                const color = this.lines.color ?? this.lines.colors?.[i] ? this.lines.colors?.[i] : '0,128,0';
+                const opacity = this.lines.opacity ?? this.lines.opacities?.[i] ? this.lines.opacities?.[i] : 1;
+                this.ctx.strokeStyle = `rgba(${color},${opacity})`;
```

Comment:
Why would we create this string in the inner-most loop, when most likely it's the same color and the same opacity?

Reply:
Changed - creating strokeStyle inside loop only in case of individual line styles